include/aspect/utilities.h

```diff
@@ -955,6 +956,63 @@ namespace aspect
                               const SymmetricTensor<2,dim> &dviscosities_dstrain_rate,
                               const double safety_factor);
 
+    /**
+     * A class that represents a binary operator between two doubles. The type of
+     * operation is specified on construction time, and can be checked later
+     * by using the operator ==. The operator () executes the operation on two
+     * double parameters and returns the result. This class is helpful for
+     * user specified operations that are not known at compile time.
+     */
+    class Operator
+    {
+      public:
+        /**
+         * An enum of supported operations.
+         */
+        enum operation
+        {
+          add,
+          subtract,
+          minimum,
+          maximum,
+          uninitialized
+        };
+
+        /**
+         * The default constructor creates an invalid operation that will fail
+         * if ever executed.
+         */
+        Operator();
+
+        /**
+         * Construct the selected operator.
+         */
+        Operator(const operation op);
+
+        /**
+         * Execute the selected operation with the given parameters and
+         * return the result.
+         */
+        double operator() (double x, double y) const;
+
+        /**
+         * Return the comparison result between the current operation and
+         * the one provided as argument.
+         */
+        bool operator== (const operation op) const;
+
+      private:
+        /**
+         * The selected operation of this object.
+         */
+        operation op;
```

Comment:
If I make this `const` I need to provide at least another `operator =` because the default move assignment operator does not work anymore. I do not know what else will be needed, so I kept it non-const for now. I also thought that it might be nice to provide a way to switch the operation later (by adding a set- function), but I am not feeling strongly about it, it was just simpler to implement this way.